batchflow/opensets/pascal.py

```diff
@@ -0,0 +1,179 @@
+    """
+    task = 'Segmentation'
+    name_classes = ['background', 'aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus', 'car', 'cat', 'chair',
+                    'cow', 'diningtable', 'dog', 'horse', 'motorbike', 'person', 'pottedplant', 'sheep', 'sofa',
+                    'train', 'tvmonitor', 'void']
+
+    def _is_train_mask(self, member, train_ids):
+        return basename(member.name).split('.')[0] in train_ids \
+               and basename(dirname(member.name)) == 'SegmentationClass'
+
+    def _is_test_mask(self, member, test_ids):
+        return basename(member.name).split('.')[0] in test_ids \
+               and basename(dirname(member.name)) == 'SegmentationClass'
+
+    def download(self, path):
+        localname = self.download_archive(path)
+        with tarfile.open(localname, "r") as archive:
+            files_in_archive = archive.getmembers()
+            train_ids = self._get_ids(archive, 'train')
+            test_ids = self._get_ids(archive, 'val')
+
+            train_images = np.array([self._extract(archive, file) for file in files_in_archive
+                                     if self._is_train_image(file, train_ids)], dtype=object)
+            train_masks = np.array([self._extract(archive, file) for file in files_in_archive
+                                    if self._is_train_mask(file, train_ids)], dtype=object)
+
+            test_images = np.array([self._extract(archive, file) for file in files_in_archive
+                                    if self._is_test_image(file, train_ids)], dtype=object)
+            test_masks = np.array([self._extract(archive, file) for file in files_in_archive
+                                   if self._is_test_mask(file, test_ids)], dtype=object)
+
+            self._train_index = DatasetIndex(np.arange(len(train_images)))
+            self._test_index = DatasetIndex(np.arange(len(test_images)))
+
+            return (train_images, train_masks), (test_images, test_masks)
+
+
+class PascalClassification(BasePascal):
+    """ Contains 11540 images and corresponding classes
+
+    Notes
+    -----
+    - Labels are provided by the vector size 20. '1' stands for the presence atleast of one object from
+    coresponding class on the image. '-1' stands for the absence. '0' indicates that the object is presented,
+    but can hardly be detected.
+     """
+    task = 'Main'
+    name_classes = ['aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus', 'car', 'cat', 'chair', 'cow',
+                    'diningtable', 'dog', 'horse', 'motorbike', 'person', 'pottedplant', 'sheep', 'sofa', 'train',
+                    'tvmonitor']
+
+    def _is_class_file(self, file):
+        """ Whether file contains labels for specific class. 20 files total. """
+        return basename(dirname(file.name)) == self.task and '_trainval.txt' in file.name
+
+    def download(self, path):
+
+        def _name(file):
```

Comment:
You use this code twice in `BasePascal`. Maybe it would be better to move it to parent class?